external-crates/move/documentation/book/src/method-syntax.md

```diff
@@ -0,0 +1,442 @@
+`Cup`, but instead we can use a `use fun` alias
+
+```move
+    fun double_double(c: Cup<u64>): (Cup<u64>, Cup<u64>) {
+        use fun b::example::double as Cup.dub;
+        (c.dub(), c.dub()) // resolves to b::example::double in both calls
+    }
+```
+
+While a `use fun` can be made in any scope, the target `<function>` of the `use fun` must have a
+first argument that is the same as the `<type>`.
+
+```move
+public struct X() has copy, drop, store;
+
+fun new(): X { X() }
+fun flag(flag: bool): u8 { if (flag) 1 else 0 }
+
+use fun new as X.new; // ERROR!
+use fun flag as X.flag; // ERROR!
+// Neither `new` nor `flag` has first argument of type `X`
+```
+
+But any first argument of the `<type>` can be used, including references and mutable references
+
+```move
+public struct X() has copy, drop, store;
+
+public fun by_val(_: X) {}
+public fun by_ref(_: &X) {}
+public fun by_mut(_: &mut X) {}
+
+// All 3 valid, in any scope
+use fun by_val as X.v;
+use fun by_ref as X.r;
+use fun by_mut as X.m;
+```
+
+Note for generics, the methods are associated for _all_ instances of the generic type. You cannot
+overload the method to resolve to different functions depending on the instantiation.
+
+```move
+public struct Cup<T>(T) has copy, drop, store;
+
+public fun value<T: copy>(c: &Cup<T>): T {
+    c.0
+}
+
+use fun value as Cup<bool>.flag; // ERROR!
+use fun value as Cup<u64>.num; // ERROR!
+// In both cases, `use fun` aliases cannot be generic, they must work for all instances of the type
+```
+
+### `public use fun` Aliases
+
+Unlike a traditional [`use`](uses.md), the `use fun` statement can be made `public`, which allows it
+to be used outside of its declared scope. A `use fun` can be made `public` if it is declared in the
+module that defines the receivers type, much like the method aliases that are
+[automatically created](#functions-in-the-defining-module) for functions in the defining module. Or
+conversely, one can think that an implicit `public use fun` is created automatically for every
+function in the defining module that has a first argument of the receiver type (if it is defined in
+that module). Both of these views are equivalent.
+
+```move
+module a::cup {
+    public struct Cup<T>(T) has copy, drop, store;
+
+    public use fun cup_borrow as Cup.borrow;
+    public fun cup_borrow<T>(c: &Cup<T>): &T {
+        &c.0
+    }
+}
+```
+
+In this example, a public method alias is created for `a::cup::Cup.borrow` and
+`a::cup::Cup.cup_borrow`. Both resolve to `a::cup::cup_borrow`. And both are "public" in the sense
+that they can be used outside of `a::cup`, without an additional `use` or `use fun`.
+
+```move
+module b::example {
+
+    fun example<T: drop>(c: a::cup::Cup<u64>) {
+        c.borrow(); // resolves to a::cup::cup_borrow
+        c.cup_borrow(); // resolves to a::cup::cup_borrow
+    }
+}
+```
+
+`public use fun`s thus serve as a way of renaming a function if you need wish to give it a cleaner
```

Comment:
```suggestion
The `public use fun` declarations thus serve as a way of renaming a function if you want to give it a cleaner
```
Not sure if this is the right verbiage, but "`public use fun`s"  just looks kinda weird.